tests/test_api_v1.py

```diff
@@ -1664,3 +1664,80 @@ def test_get_environment_variables(app, client, db, worker_auth_env, env_vars):
+    ("logs_content", "stale", "finalized", "expected"),
+    (
+        ("foobar", False, False, {"status": 200, "mimetype": "text/plain", "data": "foobar"}),
+        ("foobar", True, False, {"status": 200, "mimetype": "text/plain", "data": "foobar"}),
+        ("", False, False, {"status": 200, "mimetype": "text/plain", "data": ""}),
+        ("", True, False, {"status": 200, "mimetype": "text/plain", "data": ""}),
+        (None, False, False, {"status": 200, "mimetype": "text/plain", "data": ""}),
+        (
+            None,
+            True,
+            False,
+            {"status": 410, "mimetype": "application/json", "json": {"error": mock.ANY}},
+        ),
+        (
+            None,
+            False,
+            True,
+            {"status": 404, "mimetype": "application/json", "json": {"error": mock.ANY}},
+        ),
+    ),
+)
+def test_get_request_logs(
+    app, client, db, worker_auth_env, tmpdir, logs_content, stale, finalized, expected
+):
+    data = {
+        "repo": "https://github.com/namespace/project.git",
+        "ref": "c50b93a32df1c9d700e3e80996845bc2e13be848",
+        "pkg_managers": ["gomod"],
+    }
+    with app.test_request_context(environ_base=worker_auth_env):
+        request = Request.from_json(data)
+    request.add_state("in_progress", "Starting things up!")
+    db.session.commit()
+
+    client.application.config["CACHITO_REQUEST_FILE_LOGS_DIR"] = str(tmpdir)
+    if finalized:
+        request.add_state("complete", "The request is complete")
+        db.session.commit()
+    if stale:
+        request.add_state("stale", "The request is stale")
+        db.session.commit()
+    request_id = request.id
+    if logs_content is not None:
+        tmpdir.join(f"{request_id}.log").write(logs_content)
+    rv = client.get(f"/api/v1/requests/{request_id}/logs")
+    assert rv.status_code == expected["status"]
+    assert rv.mimetype == expected["mimetype"]
+    if "data" in expected:
+        assert rv.data.decode("utf-8") == expected["data"]
+    if "json" in expected:
+        assert rv.json == expected["json"]
+
+
+def test_get_request_logs_not_configured(app, client, db, worker_auth_env):
+    data = {
+        "repo": "https://github.com/namespace/project.git",
+        "ref": "c50b93a32df1c9d700e3e80996845bc2e13be848",
+        "pkg_managers": ["gomod"],
+    }
+    with app.test_request_context(environ_base=worker_auth_env):
+        request = Request.from_json(data)
+    request.add_state("in_progress", "Starting things up!")
+    db.session.commit()
+
+    client.application.config["CACHITO_REQUEST_FILE_LOGS_DIR"] = None
+    request_id = request.id
+    rv = client.get(f"/api/v1/requests/{request_id}/logs")
+    assert rv.status_code == 404
+    assert rv.mimetype == "application/json"
+    assert rv.json == {"error": "The requested resource was not found"}
+
+    rv = client.get(f"/api/v1/requests/{request_id}")
+    assert rv.status_code == 200
+    assert "logs" not in rv.json
```

Comment:
Should we also test the opposite case, when the logs are present in the request response?